models/maat/R/model2netcdf.MAAT.R

```diff
@@ -86,128 +98,116 @@ model2netcdf.MAAT <- function(rundir, outdir, sitelat = -999, sitelon = -999, st
+                           newunits="kg H2O m-2 s-1", missval=-999, longname="Leaf Stomatal Conductance", ncdims=ncdims)
+      output <- var_update(maat.output$ci, output, "ci", "Ci", oldunits="Pa",
+                           newunits="Pa", missval=-999, longname="Leaf Internal CO2 Concentration", ncdims=ncdims)
+      output <- var_update(maat.output$cc, output, "cc", "Cc", oldunits="Pa",
+                           newunits="Pa", missval=-999, longname="Leaf Mesophyll CO2 Concentration", ncdims=ncdims)
+      
+      ## put in time_bounds before writing out new nc file
+      output$var[[length(output$var) + 1]] <- ncdf4::ncvar_def(name="time_bounds", units='', 
+                                                               longname = "history time interval endpoints", 
+                                                               dim=list(time_interval,time = t), 
+                                                               prec = "double")
+      output$dat[[length(output$dat) + 1]] <- c(rbind(bounds[, 1], bounds[, 2]))
+      ## !!TODO: ADD MORE MAAT OUTPUTS HERE!! ##  
     }
-
-    ncdims <- list(lon, lat, t)
-
-    ## Subset data for processing
-    # setup netCDF time variable for year
-    time_interval <- ncdf4::ncdim_def(name = "hist_interval", 
-                                      longname = "history time interval endpoint dimensions",
-                                      vals = 1:2, units="")
-
-    #output <- purrr::pmap(process_maat_variable, process_tbl, missval = -999, ncdims = ncdims)
-    output <- purrr::pmap(process_tbl, process_maat_variable, missval = -999, ncdims = ncdims)
-
-    output <- c(output, list(list(
-      var = ncdf4::ncvar_def(name = "time_bounds", units = "", 
-                             longname = "history time interval endpoints", 
-                             dim = list(time_interval, time = t), 
-                             prec = "double"),
-      dat = c(rbind(bounds[, 1], bounds[, 2]))
-    )))
     
     ## write netCDF data
-    nc_vars <- purrr::map(output, "var") # Extract var from each output
-    ncout <- ncdf4::nc_create(file.path(outdir, paste(year, "nc", sep = ".")), nc_vars)
-    on.exit(ncdf4::nc_close(ncout))
+    ncout <- ncdf4::nc_create(file.path(outdir, paste(year, "nc", sep = ".")),output$var)
     ncdf4::ncatt_put(ncout, "time", "bounds", "time_bounds", prec=NA)
-    for (output_var in output) {
+    for (i in seq_along(output$var)) {
       #print(i)  # for debugging
-      ncdf4::ncvar_put(ncout, output_var$var, output_var$dat)
+      ncdf4::ncvar_put(ncout, output$var[[i]], output$dat[[i]])
     }
     
     ## extract variable and long names to VAR file for PEcAn vis
-    utils::write.table(
-      sapply(ncout$var, function(x) { x$longname }), 
-      file = file.path(outdir, paste(year, "nc.var", sep = ".")), 
-      col.names = FALSE, 
-      row.names = TRUE, 
-      quote = FALSE
-    )
+    utils::write.table(sapply(ncout$var, function(x) { x$longname }), 
```

Comment:
No need to change, but FYI, in `purrr`, the `sapply` statement can be written succinctly as:

```r
purrr::map_chr(ncout$var, "longname")
```

Also, if you want to stick with base R, I would generally recommend this, which is type-safe and faster:

```r
# Args to vapply are:
# 1. List
# 2. Function to call
# 3. The kind and size of object returned by (2) -- in this case, length 1 character
vapply(ncout$var, function(x) x[["longname"]], character(1))
```